### PR TITLE
Get Application Gateway Server Variables APIs

### PR DESCRIPTION
**Pull request by nirkum18**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json [hunk 1]
```diff
@@ -2046,6 +2136,42 @@
         "ruleGroupName"
       ],
       "description": "Allows to disable rules within a rule group or an entire rule group."
+    },
+	"ApplicationGatewayAvailableServerVariablesResult": {
```
Comment (amarzavery) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: please add "type": "object" to all the model definitions.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json [hunk 2]
```diff
@@ -451,6 +451,96 @@
         },
         "x-ms-long-running-operation": true
       }
+    },
+	"/subscriptions/{subscriptionId}/providers/Microsoft.Network/applicationGatewayAvailableServerVariables": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: these resource type names are pretty long, can they be condensed without losing meaning?

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json [hunk 3]
```diff
@@ -451,6 +451,96 @@
...
+        "description": "Lists all available server variables.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: there should be a "default" response with error schema

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json [hunk 4]
```diff
@@ -451,6 +451,96 @@
...
+          "200": {
+            "description": "Success. The operation returns a list of all available request headers.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationGatewayAvailableRequestHeadersResult"
+            }
```
Comment (KrisBash) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Same as prior, please add a "default" error response

**Review by KrisBash — CHANGES_REQUESTED** [review]
Please see comments

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 3 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: We have only one response for this API. So it is default only. Please let me know if i'm missing anything.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 4 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: We have only one response for this API. So it is by default "default".

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 2 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Actually NRP dev code for these API's has been checked in with these resource type names. So i think for now we can keep these names so as to not to break the existing changes.

Comment (KrisBash) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: ok

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 3 above)
Comment (KrisBash) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Will this API never return an error? The "default" response is a catch-all for non-success responses:
```         "default": {
	            "description": "Error response describing why the operation failed.",
	            "schema": {
	              "$ref": "#/definitions/CloudError"
	            }
	          }
```

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 4 above)
Comment (KrisBash) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: see comment above

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 1 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Added the "type".

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 4 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Added the "default" error response

File: specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json (hunk 3 above)
Comment (nirkum18) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-11-01/applicationGateway.json]: Added the "default" error message.

**Review by amarzavery — APPROVED** [review]

**Review by amarzavery — APPROVED** [review]

**Review by KrisBash — APPROVED** [review]
LGTM , thanks

**Comment by amarzavery** [comment]
@fearthecowboy @lmazuel @daschult @sergey-shandar  - The Travis CI job to generate node and java sdk is failing. The java job fails because it is taking more than 30 mins to generate network sdk and the node job fails because it is not doing anything after installing the autorest.modeler.

May be configuring the travis.yml file to use 10.x version of node.js may fix the issue. 

Do you guys think there could be anything else that could be going wrong?

Please take a look at the java and node checks. They will have links to travis ci logs.

**Comment by lmazuel** [comment]
@amarzavery Yes, SwaggerToSdk still is using NodeJS 8.x. I try to update it and see what happen. Thanks!

**Pull request merged**